### PR TITLE
HW-09: Node Chokidar

### PR DESCRIPTION
**Pull request by vbarbasins**
# Homework 09-node-chokidar

dirWatcher watches file name changes and emits event with added/removed files (includes renaming)

importer listens to the event and tries to get new data from files which was included into the event: 
- about file deletion notified in console
- data from added files emitted with new event

added for demo: app listens to the importer event emition and consoles out received objects

### CODE REVIEW

File: homeworks/09-node-chokidar/app.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import { DirWatcher, DirWatcherConfig } from './modules/dirWatcher';
+import { Importer, IMPORTED_EVENT } from './modules/importer';
+
+const PATH_TO_DATA = './data';
+
+const DIR_WATCHER_CONFIG: DirWatcherConfig = {
+  dirPath: PATH_TO_DATA,
+  fileExtention: 'CSV',
+  watchInterval: 10000,
```
Comment (vladislavkovaliov) [inline homeworks/09-node-chokidar/app.ts]:
```
watchInterval: process.env.INTERVAL || 10000,
```
i do think that it'd be nice to have possibility to set an interval via env arguments.

File: homeworks/09-node-chokidar/modules/dirWatcher.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  fileExtention: string,
+  watchInterval: number,
+};
+
+export class DirWatcher {
+  private watchedFileNames: string[];
+
+  constructor(
+    public eventEmitter: EventEmitter,
+    public config: DirWatcherConfig,
```
Comment (vladislavkovaliov) [inline homeworks/09-node-chokidar/modules/dirWatcher.ts]: i guess that providing getter for config would be better that it's public. It'd be great to make the config only for reading.

File: homeworks/09-node-chokidar/modules/importer.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      .catch(() => {
+        console.log(`File removed at ${path}`);
+        return [];
+      })
+  );
+
+  private importSync = (path: string): Object[] => {
+    let data;
+    try {
+      data = readFileSync(path, 'utf8');
```
Comment (vladislavkovaliov) [inline homeworks/09-node-chokidar/modules/importer.ts]: i guess that defining variable `data` should be inside try block

**Comment by vladislavkovaliov** [comment]
no major issues

File: homeworks/09-node-chokidar/app.ts (hunk 1 above)
Comment (vbarbasins) [reply homeworks/09-node-chokidar/app.ts]: added in https://github.com/vbarbasins/evo-ts-bootcamp/pull/10/commits/9e1253a8620799f26e2af37c0a2741153355ecd4

File: homeworks/09-node-chokidar/modules/dirWatcher.ts (hunk 2 above)
Comment (vbarbasins) [reply homeworks/09-node-chokidar/modules/dirWatcher.ts]: added in https://github.com/vbarbasins/evo-ts-bootcamp/pull/10/commits/37e95faf8f9b2447ff9b0b2293cec8e82c681932

File: homeworks/09-node-chokidar/modules/importer.ts (hunk 3 above)
Comment (vbarbasins) [reply homeworks/09-node-chokidar/modules/importer.ts]: move in https://github.com/vbarbasins/evo-ts-bootcamp/pull/10/commits/80a2d5181d37a9ab51cc552a661b6dbe7c257fb9

**Pull request merged**